pkg/binlog/position.go

```diff
@@ -316,6 +316,39 @@ func CompareLocation(location1, location2 Location, cmpGTID bool) int {
 	return compareIndex(location1.Suffix, location2.Suffix)
 }
 
+// CompareLocationAsPossible returns:
+//   1 if point1 is bigger than point2
+//   0 if point1 is equal to point2
+//   -1 if point1 is less than point2
+// The difference is that this function will compare positions if gtid sets are both empty.
+func CompareLocationAsPossible(location1, location2 Location, cmpGTID bool) int {
+	if cmpGTID {
+		cmp, canCmp := CompareGTID(location1.gtidSet, location2.gtidSet)
+		if canCmp {
+			if cmp != 0 {
+				return cmp
+			}
+			if cmp = compareIndex(location1.Suffix, location2.Suffix); cmp != 0 {
+				return cmp
+			}
+			if location1.gtidSet != nil && location1.gtidSet.String() != "" {
+				return cmp
+			}
```

Comment:
What if both gtid sets are empty, but suffix are not? e.g. location1:(pos1,"",1), location2:(pos2,"",0)

Reply:
I think `Suffix` is only used in `handle-error`?